ddtrace/tracer/textmap.go

```diff
@@ -114,74 +115,188 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
 	if cfg.PriorityHeader == "" {
 		cfg.PriorityHeader = DefaultPriorityHeader
 	}
-	return &propagator{cfg}
+	return &propagator{
+		injectors:  makeInjectors(cfg),
+		extractors: makeExtractors(cfg),
+	}
 }
 
-// propagator implements a propagator which uses TextMap internally.
+// propagator implements a Propagator that supports TextMap carriers.
 // It propagates the trace and span IDs, as well as the baggage from the
 // context.
-type propagator struct{ cfg *PropagatorConfig }
+type propagator struct {
+	injectors  []textMapPropagator
+	extractors []textMapPropagator
+}
+
+// makeInjectors returns a list of injectors to apply when propagating a
+// context. By default, only the datadog propagation style will be used.
+// If the DD_PROPAGATION_STYLE_INJECT environment variable is set,
+// this can override the default behavior.
+func makeInjectors(cfg *PropagatorConfig) []textMapPropagator {
+	dd := &datadogPropagator{cfg}
```

Comment:
We don't need to make both of these structures ahead, because we don't know if we'll need both (at least the B3 one).